testrunner-running/src/main/scala/edu/illinois/cs/statecapture/StateCapture.java

```diff
@@ -0,0 +1,755 @@
+                    } catch (NoClassDefFoundError NCDFE) {
+                        continue;
+                    }
+                }
+                Class[] arrayClasses = new Class[list.size()];
+                list.toArray(arrayClasses);
+                loadedClasses = arrayClasses;
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+        for (Class c : loadedClasses) {
+            // Ignore classes in standard java to get top-level
+            // TODO(gyori): make this read from file or config option
+            String clz = c.getName();
+            if ((clz.contains("java.") && !clz.startsWith("java.lang.System"))
+                    || (clz.contains("javax.") && !clz.startsWith("javax.cache.Caching"))
+                    || clz.contains("javafx.")
+                    || clz.contains("jdk.")
+                    || clz.contains("scala.")
+                    || clz.contains("sun.")
+                    || clz.contains("edu.illinois.cs")
+                    || clz.contains("org.custommonkey.xmlunit")
+                    || clz.contains("org.junit")
+                    || clz.contains("statecapture.com.")
+                    || clz.contains("statecapture.org.")
+                    || clz.equals("com.openpojo.reflection.impl.AClassWithBadMethod__Generated_OpenPojo")) {
+                continue;
+            }
+
+            Set<Field> allFields = new HashSet<Field>();
+            try {
+                Field[] declaredFields = c.getDeclaredFields();
+                Field[] fields = c.getFields();
+                allFields.addAll(Arrays.asList(declaredFields));
+                allFields.addAll(Arrays.asList(fields));
+            } catch (NoClassDefFoundError e) {
+                e.printStackTrace();
+                continue;
+            } catch (Exception exception) {
+                exception.printStackTrace();
+                continue;
+            }
+            // prepare for the subxml fold
+
+            for (Field f : allFields) {
+                String fieldName = getFieldFQN(f);
+
+                if (ignores.contains(fieldName)) {
+                    continue;
+                }
+
+                // if a field is final and has a primitive type there's no point to capture it.
+                if (Modifier.isStatic(f.getModifiers())
+                        && !(Modifier.isFinal(f.getModifiers()) &&  f.getType().isPrimitive())) {
+                    try {
+                        if (shouldCapture(f)) {
+                            allFieldName.add(fieldName);
+                        }
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                        continue;
+                    }
+                }
+            }
+        }
+
+        int num = countDirNums(subxmlFold) - 1;
+        PrintWriter writer = new PrintWriter(MainAgent.fieldFold + "/" + num + ".txt", "UTF-8");
```

Comment:
Let's not just increment some counter and use it as the name, let's give it some constant name for the output.